st2common/st2common/content/loader.py

```diff
@@ -268,3 +269,137 @@ def _load(self, parser_func, file_path):
             except ParserError:
                 LOG.exception("Failed loading content from %s.", file_path)
                 raise
+
+
+class OverrideLoader(object):
+    """
+    Class for loading pack override data
+    """
+
+    # Mapping of permitted override types to resource name
+    ALLOWED_OVERRIDE_TYPES = {
+        "sensors": "class_name",
+        "actions": "name",
+        "rules": "name",
+        "aliases": "name",
+    }
+
+    ALLOWED_OVERRIDE_NAMES = [
+        "enabled",
+    ]
+
+    DEFAULT_OVERRIDE_VALUES = {"enabled": True}
+
+    def override(self, pack_name, resource_type, content):
+
+        """
+        Loads override content for pack, and updates content
+
+        :param pack_name: Name of pack
+        :type pack_name: ``str``
+        :param resource_type: Type of resource loading
+        :type type: ``str``
+        :param content: Content as loaded from meta information
+        :type content: ``object``
+        :return: Whether data was overridden
+        :rtype: ``bool``
+
+
+        """
+        orig_content = content.copy()
+        if resource_type not in self.ALLOWED_OVERRIDE_TYPES.keys():
+            raise ValueError(
+                f"Invalid override type of {resource_type} attempted for pack {pack_name}"
+            )
+
+        override_dir = os.path.join(cfg.CONF.system.base_path, "overrides")
+        # Apply global overrides
+        global_file = os.path.join(override_dir, "global.yaml")
```

Comment:
I wonder if we should do something during pack install to make sure the pack name is not `global` so people don't shoot themselves in the foot.

Reply:
:100: Nice find of the corner case

Reply:
@cognifloyd I'll take a look at how best to do that, the alternative is to make the global something we can't use as a packname...